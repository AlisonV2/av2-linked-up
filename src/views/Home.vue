<template>
  <div class="container">
    <HomeGeoloc />
    <HomeCategories />
  </div>
</template>

<script>
import HomeGeoloc from '@/components/home/Geoloc';
import HomeCategories from '@/components/home/Categories';

/**
 * @type Page
 * @name Home
 */
export default {
  name: 'Home',
  components: {
    HomeCategories,
    HomeGeoloc
  },
};
</script>
