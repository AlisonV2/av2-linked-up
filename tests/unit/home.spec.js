import { shallowMount } from '@vue/test-utils';
import Home from '@/views/Home.vue';
import HomeGeoloc from '@/components/home/Geoloc';
import HomeCategories from '@/components/home/Categories';

/**
 * @module HomePageTest
 */
describe('HomeGeoloc.vue', () => {
  /**
   * Check if HomeGeoloc is rendered
   */
  it('Check if HomeGeoloc is rendered', () => {
    const component = shallowMount(Home, {
      global: {
        components: {
          HomeCategories,
          HomeGeoloc,
        },
      },
    });
    const geoloc = component.findComponent(HomeGeoloc);
    expect(geoloc.exists()).toBe(true);
  });

  /**
   * Check if HomeCategories is rendered
   */
  it('Check if HomeCategories is rendered', () => {
    const component = shallowMount(Home, {
      global: {
        components: {
          HomeCategories,
          HomeGeoloc,
        },
      },
    });
    const categories = component.findComponent(HomeCategories);
    expect(categories.exists()).toBe(true);
  });
});
