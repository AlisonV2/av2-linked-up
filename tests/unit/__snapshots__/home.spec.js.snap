// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue HomePage Snapshot 1`] = `
<div
  class="container"
>
  <home-geoloc-stub />
  <home-categories-stub />
</div>
`;
